scipy/ndimage/src/_rank_filter_1d.cpp

```diff
@@ -155,7 +155,26 @@ void _rank_filter(T *in_arr, int rank, int arr_len, int win_len, T *out_arr,
                   int mode, T cval, int origin) {
   int i, arr_len_thresh, lim = (win_len - 1) / 2 - origin;
   int lim2 = arr_len - lim;
-  if (lim2 < 0) return;
+  if (lim2 < 0) {
+      if (arr_len == 1) {
```

Comment:
There is no `else` clause here, which looks off. Is it impossible, missing, or did you intend for that case to fall through now to beyond line 177 in this diff, rather than hit `return;`?

Reply:
I've purged the `arr_len` check and the full suite still passes locally.

That said, I had intended for the last thing you mention, I really only wanted to intercept the size-1 array case and the code inside the conditional was intended for the simple scenario that was failing in the original ticket. `_filters.py` also has the size-1 exception/allowance for the fast path.  But I can't prove with any testing that we need that extra guard anymore, so while that bothers me, I've removed it.

I did double check that we still need the broader `if (lim2 < 0)` guard--there are gruesome segfaults locally without that.

Reply:
That's why I had it originally, but we can't prove a need for the guard with a public function test right? So this would just be a preference to have a private function behave more robustly on its own?

Reply:
alright, I pushed that up